src/client/datascience/variablesView/notebookWatcher.ts

```diff
@@ -40,7 +41,11 @@ export class NotebookWatcher implements INotebookWatcher {
         return this._onDidRestartActiveNotebook.event;
     }
     public get activeNotebook(): INotebook | undefined {
-        return this.notebookEditorProvider.activeEditor?.notebook || this.getActiveInteractiveWindowNotebook();
+        return (
+            this.notebookEditorProvider.activeEditor?.notebook ||
+            this.getActiveInteractiveWindowNotebook() ||
+            this.getActiveDataViewerNotebook()
```

Comment:
This line here is what enables the whole thing. The 'active' notebook can be gathered even from the dataviewer that's active.

Reply:
Wouldn't the same apply to plot viewers?
I think we should just show the Title of the notebook in the variable view & never hide the list of variables, then we'd never have to fix such issues one by one.

Reply:
I'm not convinced this is a good UX, i think we should not hide the variables (seems like a never ending list of scenarios we need to cover), lets discuss in standup.

It should be similar to debugger variables, we display debugger varaibles always, unbless you switch to another debug session.

Reply:
Variables don't work without a notebook though (plotviewer or file not associated with an interactive window would mean it was broken). Prior to this change the variable list would sometimes clear and sometimes just not work (if you scrolled or tried to open a dataframe it would just do nothing).

This is forcing it to refresh every time you switch active editors so that it always reflects the state of the active editor.